tests/static-analysis/TestUsingMocks.php

```diff
@@ -0,0 +1,113 @@
+<?php declare(strict_types=1);
```

Comment:
Will need to add a CI step to verify this. Also unsure if we should split it into multiple type-check `.phpt` scenarios